content/recipes/nestia.md

```diff
@@ -0,0 +1,698 @@
+npx nestia sdk "src/controller" \ 
+    --exclude "src/controller/test" \
+    --out "src/api"
+
+# BUILDING SWAGGER.JSON IS ALSO POSSIBLE
+npx nestia swagger "src/controller" -- out "swagger.json"
+```
+
+Don't write any swagger comment and DTO decorator. Just run the `nestia` up.
+
+  - No swagger comment/decorator
+  - No DTO comment/decorator
+  - Only pure `interface`s and NestJS code are required
+    - Support generic typed `interface`s and `controller`s
+    - Support union/intersection types
+    - Support conditional types
+
+The `nestia` is an evolved automatic `SDK` and `Swagger` generator than ever, who does not require any type of swagger comment or DTO decorator function, by analyzing your `NestJS` developed backend server code in the compilation level (`nestia` utilizes the TypeScript Compiler API).
+
+Therefore, don't write any swagger comment and don't use any DTO related decorator function. Just use the pure interface type with the pure `NestJS` code. Reading below sections and looking at example codes of the `nestia`, feel how the `nestia` is much stronger than the legacy `@nestjs/swagger`.
+
+<img src="https://github.com/samchon/nestia/wiki/images/sdk-comparison-table.png" style="max-width: 100%" />
+
+```typescript
+// IMPORT SDK LIBRARY WHO'VE BEEN GENERATED BY THE NESTIA
+import api from "@samchon/shopping-api";
+import { IPage } from "@samchon/shopping-api/lib/structures/IPage";
+import { ISale } from "@samchon/shopping-api/lib/structures/ISale";
+import { ISaleArticleComment } from "@samchon/shopping-api/lib/structures/ISaleArticleComment";
+import { ISaleQuestion } from "@samchon/shopping-api/lib/structures/ISaleQuestion";
+
+export async function trace_sale_question_and_comment
+    (connection: api.IConnection): Promise<void>
+{
+    // LIST UP SALE SUMMARIES
+    const index: IPage<ISale.ISummary> = await api.functional.shoppings.sales.index
+    (
+        connection,
+        "general",
+        { limit: 100, page: 1 }
+    );
+
+    // PICK A SALE
+    const sale: ISale = await api.functional.shoppings.sales.at
+    (
+        connection, 
+        index.data[0].id
+    );
+    console.log("sale", sale);
+
+    // WRITE A QUESTION
+    const question: ISaleQuestion = await api.functional.shoppings.sales.questions.store
+    (
+        connection,
+        "general",
+        sale.id,
+        {
+            title: "How to use this product?",
+            body: "The description is not fully enough. Can you introduce me more?",
+            files: []
+        }
+    );
+    console.log("question", question);
+
+    // WRITE A COMMENT
+    const comment: ISaleArticleComment = await api.functional.shoppings.sales.comments.store
+    (
+        connection,
+        "general",
+        sale.id,
+        question.id,
+        {
+            body: "p.s) Can you send me a detailed catalogue?",
+            anonymous: false
+        }
+    );
+    console.log("comment", comment);
+}
+```
```

Comment:
```suggestion

#### Generated SDK Use Example

The following is an example of what the use of the generated SDK might look like. The package name `@samchon/shopping-api` is the name of the generated SDK.

```typescript
import api from "@samchon/shopping-api";
import { IPage } from "@samchon/shopping-api/lib/structures/IPage";
import { ISale } from "@samchon/shopping-api/lib/structures/ISale";
import { ISaleArticleComment } from "@samchon/shopping-api/lib/structures/ISaleArticleComment";
import { ISaleQuestion } from "@samchon/shopping-api/lib/structures/ISaleQuestion";
export async function trace_sale_question_and_comment
    (connection: api.IConnection): Promise<void>
{
    // LIST UP SALE SUMMARIES
    const index: IPage<ISale.ISummary> = await api.functional.shoppings.sales.index
    (
        connection,
        "general",
        { limit: 100, page: 1 }
    );
    // PICK A SALE
    const sale: ISale = await api.functional.shoppings.sales.at
    (
        connection, 
        index.data[0].id
    );
    console.log("sale", sale);
    // WRITE A QUESTION
    const question: ISaleQuestion = await api.functional.shoppings.sales.questions.store
    (
        connection,
        "general",
        sale.id,
        {
            title: "How to use this product?",
            body: "The description is not fully enough. Can you introduce me more?",
            files: []
        }
    );
    console.log("question", question);
    // WRITE A COMMENT
    const comment: ISaleArticleComment = await api.functional.shoppings.sales.comments.store
    (
        connection,
        "general",
        sale.id,
        question.id,
        {
            body: "p.s) Can you send me a detailed catalogue?",
            anonymous: false
        }
    );
    console.log("comment", comment);
}
```